### DISCUSSION TITLE
Tomcat 10 compatibility

### DISCUSSION THREAD

**Discussion by shoemael** [OWASP/www-project-csrfguard#131] [Ideas]
Hello.  Is there a distribution based on the newer jakarta.servlet.* package naming?  Tomcat 10 has moved to Jakarta EE and the csrf listeners/filters appear to be incompatible.  They cause unknown class exceptions for javax.servlet.ServletInputStream, etc.

I noticed another question regarding Jakarta EE which was said to be the first.  So here's a ++ for support for the jakarta namespace.

**Comment by forgedhallpass** [comment]
Hello @shoemael,

Do you have any proposition on the matter? We could potentially use the [Eclipse Transformer](https://github.com/eclipse/transformer) project from maven to generate a new distribution at release time. Would you be interested in implementing it?

Related previous discussion: https://github.com/OWASP/www-project-csrfguard/discussions/41

**Reply by shoemael** [reply]
I was previously unaware of the Eclipse Transformer project.  At first glance it seems like that should work nicely.  I'll give that a shot and let you know how it works out.  If I can find the time I'll create a fork and see about a possible implementation strategy.

**Comment by forgedhallpass** [comment]
Solved under #176 
Released under version [4.3.0-jakarta](https://github.com/OWASP/www-project-csrfguard/releases/tag/4.3.0-jakarta)
Tested with Tomcat 10.1.9 and OpenJDK 11.0.19
